convert2rhel/unit_tests/actions/conversion/preserve_only_rhel_kernel_test.py

```diff
@@ -0,0 +1,430 @@
+                "/boot/loader/entries/b5aebfb91bff486bb9d44ba85e4ae683-5.4.17-2011.7.4.el8uek.x86_64.conf",
+            ],
+        )
+        monkeypatch.setattr(system_info, "version", Version(8, 0))
+        monkeypatch.setattr(system_info, "arch", "x86_64")
+        monkeypatch.setattr(
+            utils,
+            "get_file_content",
+            lambda x: "1b11755afe1341d7a86383ca4944c324\n",
+        )
+        run_subprocess_mocked = RunSubprocessMocked(
+            side_effect=unit_tests.run_subprocess_side_effect(
+                (
+                    (
+                        "/usr/sbin/grubby",
+                        "--default-kernel",
+                    ),
+                    (
+                        "bootloader",
+                        return_code_1,
+                    ),
+                ),
+                (
+                    (
+                        "/usr/sbin/grubby",
+                        "--set-default",
+                        "bootloader",
+                    ),
+                    (
+                        "bootloader",
+                        return_code_2,
+                    ),
+                ),
+            ),
+        )
+
+        monkeypatch.setattr(
+            utils,
+            "run_subprocess",
+            value=run_subprocess_mocked,
+        )
+
+        fix_invalid_grub2_entries_instance.run()
+        print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+        print(fix_invalid_grub2_entries_instance.messages)
+        assert expected.issuperset(fix_invalid_grub2_entries_instance.messages)
+        assert expected.issubset(fix_invalid_grub2_entries_instance.messages)
+
+
+class TestFixDefaultKernel:
+    @pytest.mark.parametrize(
+        ("system_name", "version", "old_kernel", "new_kernel", "not_default_kernels"),
+        (
+            (
+                "Oracle Linux Server release 7.9",
+                Version(7, 9),
+                "kernel-uek",
+                "kernel",
+                ("kernel-uek", "kernel-core"),
+            ),
+            (
+                "Oracle Linux Server release 8.1",
+                Version(8, 1),
+                "kernel-uek",
+                "kernel-core",
+                ("kernel-uek", "kernel"),
+            ),
+            (
+                "CentOS Plus Linux Server release 7.9",
+                Version(7, 9),
+                "kernel-plus",
+                "kernel",
+                ("kernel-plus",),
+            ),
+        ),
+    )
+    def test_fix_default_kernel_converting_success(
+        self,
+        system_name,
+        version,
+        old_kernel,
+        new_kernel,
+        not_default_kernels,
+        caplog,
+        monkeypatch,
+        fix_default_kernel_instance,
+    ):
+        monkeypatch.setattr(system_info, "name", system_name)
+        monkeypatch.setattr(system_info, "arch", "x86_64")
+        monkeypatch.setattr(system_info, "version", version)
```

Comment:
Probably could be changed to @all_systems?

Reply:
Tried it but got a bunch of failures